compiler/ghc/default.nix

```diff
@@ -384,13 +454,30 @@ stdenv.mkDerivation (rec {
   dontStrip = true;
   dontPatchELF = true;
   noAuditTmpdir = true;
-} // lib.optionalAttrs stdenv.buildPlatform.isDarwin {
+} // lib.optionalAttrs (stdenv.buildPlatform.isDarwin || stdenv.targetPlatform.isWindows) {
   # ghc install on macOS wants to run `xattr -r -c`
   # The macOS version fails because it wants python 2.
   # The nix version of xattr does not support those args.
   # Luckily setting the path to something that does not exist will skip the step.
-  preBuild = ''
+  preBuild = lib.optionalString stdenv.buildPlatform.isDarwin ''
     export XATTR=$(mktemp -d)/nothing
+  ''
+  # We need to point at a stand in `windows.h` header file so that the RTS headers can
+  # work on the hostPlatform.  We also need to work around case sensitve file system issues.
```

Comment:
Why can't we use the one from mingw32?

Reply:
It does not work with linux gcc.  There is a patch coming for GHC that removes the dependency on building windows.h with the host platform compiler.  We can drop this when that lands.